dandiapi/api/models/oauth.py

```diff
@@ -0,0 +1,20 @@
+import re
+from django.core.exceptions import ValidationError
+from oauth2_provider.models import AbstractApplication
+
+
+class Application(AbstractApplication):
+    # override URL validation
+    def clean(self):
+        try:
+            super().clean()
+        except ValidationError as e:
+            # don't validate URLs so we can use regexes too
+            if 'Enter a valid URL.' not in str(e):
+                raise e
+
+    def redirect_uri_allowed(self, uri):
+        for allowed_uri in self.redirect_uris.split():
+            if re.fullmatch(uri, allowed_uri):
```

Comment:
Let's see if we can use fnmatch instead https://docs.python.org/3/library/fnmatch.html#fnmatch.fnmatch

For reference: https://github.com/girder/girder/blob/d55b2c60c678f7acc12f5c54b9d60f3838fcbab6/girder/api/rest.py#L754

Reply:
FWIW reference use of `fnmatch` https://github.com/girder/girder/blob/d55b2c60c678f7acc12f5c54b9d60f3838fcbab6/girder/api/rest.py#L754

Reply:
Done.